src/main/java/net/sf/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -16,201 +16,202 @@
+        caseSensitive.setToolTipText(Localization.lang("Case sensitive"));
+        caseSensitive.addActionListener(event -> {
+            searchPreferences.setCaseSensitive(caseSensitive.isSelected());
+            performSearch();
+        });
+
+        updateSearchModeButtonText();
+        searchModeButton.addActionListener(event -> toggleSearchModeAndSearch());
 
-        JButton clearSearchButton = new JButton(IconTheme.JabRefIcon.CLOSE.getSmallIcon());
         clearSearchButton.setToolTipText(Localization.lang("Clear"));
-        clearSearchButton.addActionListener(l -> endSearch());
+        clearSearchButton.addActionListener(event -> endSearch());
 
-        this.add(clearSearchButton);
+        searchField.addFocusListener(Globals.getFocusListener());
+        searchField.addActionListener(event -> performSearch());
+        JTextFieldChangeListenerUtil.addChangeListener(searchField, e -> performSearch());
 
-        searchModeButton = new JButton();
-        updateSearchModeButtonText();
-        searchModeButton.addActionListener(l -> toggleSearchModeAndSearch());
+        String endSearch = "endSearch";
+        searchField.getInputMap().put(Globals.getKeyPrefs().getKey(KeyBinding.CLEAR_SEARCH), endSearch);
+        searchField.getActionMap().put(endSearch, new AbstractAction() {
+            @Override
+            public void actionPerformed(ActionEvent event) {
+                endSearch();
+            }
+        });
 
-        JToolBar toolBar = new OSXCompatibleToolbar();
+        autoCompleteSupport.install();
+
+        setLayout(new WrapLayout(FlowLayout.RIGHT));
         toolBar.setFloatable(false);
-        toolBar.add(clearSearchButton);
+        if (OS.OS_X) {
+            searchField.putClientProperty("JTextField.variant", "search");
+            toolBar.add(searchField);
+        } else {
+            toolBar.add(searchIcon);
+            toolBar.add(searchField);
+            toolBar.add(clearSearchButton);
+        }
+        toolBar.addSeparator();
+        toolBar.add(openCurrentResultsInDialog);
+        toolBar.add(globalSearch);
         toolBar.addSeparator();
         toolBar.add(regularExp);
         toolBar.add(caseSensitive);
         toolBar.addSeparator();
         toolBar.add(searchModeButton);
         toolBar.addSeparator();
-        toolBar.add(openCurrentResultsInDialog);
-        globalSearch = new JButton(Localization.lang("Search globally"));
-        globalSearch.setToolTipText(Localization.lang("Search in all open databases"));
-        globalSearch.addActionListener(l -> {
-            AbstractWorker worker = new GlobalSearchWorker(basePanel.frame(), getSearchQuery());
-            worker.run();
-            worker.update();
-        });
-        globalSearch.setEnabled(false);
-        toolBar.add(globalSearch);
-        toolBar.addSeparator();
         toolBar.add(new HelpAction(HelpFiles.SEARCH));
-
+        toolBar.addSeparator();
+        toolBar.add(currentResults);
```

Comment:
This will cause the toolbar to jump around when typing and clearing a search query. To fit the searchbar in the first row I need to use full screen mode and than there is no space to the right at all. So once I start typing the result text will be displayed and the searchbar jumps to the second row, when I clear the search it jumps back to the 1st row since the text is invisible. In my opinion the text with the number of found results should be somewhere else.